src/gui/AttachmentBubble.ts

```diff
@@ -28,19 +29,29 @@ export class AttachmentBubble implements Component<AttachmentBubbleAttrs> {
 
 	view(vnode: Vnode<AttachmentBubbleAttrs>): Children {
 		const { attachment } = vnode.attrs
-		const extension = getFileExtension(attachment.name)
-		const rest = getFileBaseName(attachment.name)
-		return m(Button, {
-			label: () => rest,
-			title: () => attachment.name,
-			icon: () => Icons.Attachment,
-			type: ButtonType.Bubble,
-			staticRightText: `${extension}, ${formatStorageSize(Number(attachment.size))}`,
-			click: async () => {
-				await showAttachmentDetailsPopup(this.dom!, vnode.attrs)
-				this.dom?.focus()
-			},
-		})
+		if (isTutanotaFile(attachment) && hasError(attachment)) {
```

Comment:
I'm probably missing some context. Why are we checking if it is a Tutanota file, and not showing for any attachment that has an error?

Reply:
We have 3 types of file objects:

 * DataFile: file that is in memory. Usually some file that the user just picked
 * FileReference: file that exists in temporary space on disk, picked files in mobile apps and such
 * TutanotaFile (entity is just called File but we rename it to avoid the naming issues). It is an actual entity that is stored in the database.

Out of these three only TutanotaFile is an Entity and `hasError()` requires an object to be an entity so we have to assert it is one before checking for errors.